File: src/main/java/javabot/commands/DisableOperation.java
package javabot.commands;

import java.util.ArrayList;
import java.util.List;

import javabot.BotEvent;
import javabot.Javabot;
import javabot.Message;

/**
 * Created Jan 26, 2009
 *
 * @author <a href="mailto:jlee@antwerkz.com">Justin Lee</a>
 */
public class DisableOperation extends OperationsCommand {
    @Override
    public List<Message> execute(final Javabot bot, final BotEvent event, final List<String> args) {
        final List<Message> messages = new ArrayList<Message>();
        if (args.isEmpty()) {
            bot.postMessage(new Message(event.getChannel(), event, "usage: disableOperation <name>"));
            bot.postMessage(new Message(event.getChannel(), event,
                "usage: use admin listOperations to see list of options"));
        } else {
            final String name = args.get(0);
            if(bot.removeOperation(name)) {
                bot.postMessage(new Message(event.getChannel(), event, name + " successfully disabled."));
                listCurrent(bot, event);
            } else {
                bot.postMessage(new Message(event.getChannel(), event, name + " not disabled.  Either it is not running"
                    + " or it's not a valid name.  see listOperations for details."));
            }
        }
        return messages;
    }
}

File: src/main/kotlin/javabot/commands/DisableOperation.kt
package javabot.commands

import com.antwerkz.sofia.Sofia
import com.beust.jcommander.Parameter
import com.google.inject.Inject
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao

class DisableOperation @Inject constructor(bot: Javabot, adminDao: AdminDao) : OperationsCommand(bot, adminDao) {
    @Parameter(names=arrayOf("-name"), description = "The name of the operation to disable", required = true)
    lateinit var operationName: String

    override fun execute(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        if (bot.disableOperation(operationName)) {
            responses.add(Message(event, Sofia.adminOperationDisabled(operationName)))
            listCurrent(event, responses)
        } else {
            responses.add(Message(event, Sofia.adminOperationNotDisabled(operationName)))
        }

        return responses
    }
}